src/redux/thunk/blockThunks.ts

```diff
@@ -0,0 +1,58 @@
+import { AppThunk } from '@redux/types';
+import { setLoadingBlock, setLatestBlocks } from '@redux/actions/blocksAction';
+import blockApis from '@apis/blocks';
+import { setResponseError } from '@redux/actions/responseErrorsActions';
+import { IBlock, IRawBlock } from '@utils/types/IBlocks';
+import { BLOCK_NAMESPACE } from '@redux/reducers/blockReducer';
+
+export const getLatestBlocks: () => AppThunk<Promise<void>> = (limit = 6) => async (
+  dispatch,
+  // _getState,
+): Promise<void> => {
+  try {
+    dispatch(setLoadingBlock());
+
+    const { data, timestamp } = await blockApis.getLatestBlock(limit);
+
+    dispatch(setLatestBlocks({ latestBlocks: data, timestamp }));
+  } catch (error) {
+    dispatch(setResponseError(true, error.message));
+  }
+};
+
+export const updateBlocksNewest: (_block: IRawBlock) => AppThunk<Promise<void>> = block => async (
+  dispatch,
+  getState,
+): Promise<void> => {
+  const prevBlocks = getState()[BLOCK_NAMESPACE].latestBlocks;
+  const newBlocks = new Map<string, IBlock>();
+  newBlocks.set(block.hash, {
+    id: block.hash,
+    difficulty: block.difficulty,
+    confirmations: block.confirmations,
+    height: block.height,
+    merkleRoot: block.merkleroot,
+    nextBlockHash: '',
+    nonce: block.nonce,
+    previousBlockHash: block.previousblockhash,
+    size: block.size,
+    solution: block.solution,
+    timestamp: block.time,
+    transactionCount: block.transactions.length,
+    transactions: block.transactions,
+  });
+  let i = 1;
+  prevBlocks.forEach((value, key) => {
+    if (i < 6) {
```

Comment:
Unexpected to see such condition, why < 6?

Reply:
It only takes 6 blocks for live. When socket emit new data I need to update the state in the store, latest blocks will be displayed first. We don't call api to fetch the data.